src/main/java/org/georchestra/gateway/autoconfigure/app/HeaderFiltersAutoConfiguration.java

```diff
@@ -30,21 +36,33 @@
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
-@Configuration
+@Configuration(proxyBeanMethods = false)
```

Comment:
I read the doc(https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Configuration.html#proxyBeanMethods--) , but cannot figure out what is the purpose of proxyBeansMethods yet: is it to avoid cglib instrumentation of the class (so, to avoid unecessary processing) ?

Reply:
> is it to avoid cglib instrumentation of the class (so, to avoid unecessary processing) ?
yes. You only really needed it if a `@Bean` method is calling other `@Bean` methods to get its dependencies